main/views.py

```diff
@@ -1,13 +1,14 @@
 from django.shortcuts import render, redirect
 from .forms import UserForm, UserProfileForm, UserFormUpdate
 from . import models
-from django.views.generic import (TemplateView, View, ListView)
+from django.views.generic import (TemplateView, View, ListView, DetailView, CreateView, UpdateView, DeleteView)
```

Comment:
To make it readable, use:
from django.views.generic import (
TemplateView, 
View, 
ListView, 
DetailView, 
CreateView, 
UpdateView, 
DeleteView
)
instead of:
from django.views.generic import (TemplateView, View, ListView, DetailView, CreateView, UpdateView, DeleteView)